src/pkg/packager/deploy.go

```diff
@@ -626,6 +633,7 @@ func setupState(ctx context.Context, c *cluster.Cluster, pkg v1alpha1.ZarfPackag
 	if s == nil {
 		return nil, errors.New("cluster state should not be nil")
 	}
+	s.Architecture = pkg.Metadata.Architecture
```

Comment:
I had not realized that this was a field on state. Looks like it was added at least a few years ago, but was not used. With how we're setting it here, it doesn't do anything. Another route we could take would be setting it in `InitState` and if state already exists then pulling it from the init package secret. But I think that's not necessary, at least as of now and we can delete this line.